trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/TypeHandler.java

```diff
@@ -430,6 +431,16 @@ public Result visit(final IDispatcher main, final IASTCompositeTypeSpecifier nod
 					bitFieldWidths.add(declaration.getBitfieldSize());
 				}
 			} else if (r instanceof SkipResult) { // skip ;)
+			} else if (r instanceof final TypesResult tr && tr.getCType() instanceof final CStructOrUnion su) {
+				if (su.isAnonymous()) {
+					// Flat unnamed (anonymous) struct or union by adding the field(s) to to the parent struct or union.
+					fNames.addAll(su.getFieldNames());
+					fTypes.addAll(Arrays.asList(su.getFieldTypes()));
+					bitFieldWidths.addAll(su.getBitFieldWidths());

```

Comment:
I understand the idea to flatten anonymous structs into a parent struct; and I guess, similarly, to flatten anonymous unions into a parent union.

But does it also make sense to flatten an anonymous struct into a parent union, or vice versa? A struct is a kind of product type (basically a tuple, but the fields are named), whereas a union is a sum type (only one of the values is present, not all). It seems to me that this information is lost by flattening one into the other, isn't it?

Reply:
I would just suggest to have a separate member in `CStructOrUnion` for anonymous fields and to adapt the methods (recursive lookup in anonymous fields, somehow return the fields in getters).